torchx/tracker/api.py

```diff
@@ -181,26 +181,27 @@ def build_trackers(
 ) -> Iterable[TrackerBase]:
     trackers = []
 
-    entrypoint_factories = load_group("torchx.tracker")
+    entrypoint_factories = load_group("torchx.tracker") or {}
     if not entrypoint_factories:
-        logger.warning(
-            "No 'torchx.tracker' entry_points are defined. Tracking will not capture any data."
-        )
-        return trackers
+        logger.warning("No 'torchx.tracker' entry_points are defined.")
 
     for entrypoint_key, config in entrypoint_and_config.items():
-        if entrypoint_key not in entrypoint_factories:
-            logger.warning(
-                f"Could not find `{entrypoint_key}` tracker entrypoint. Skipping..."
+        factory = entrypoint_factories.get(entrypoint_key)
+        if factory is None:
+            logger.info(
+                f"Could not find `{entrypoint_key}` among entry_points, checking modules..."
             )
-            continue
-        factory = entrypoint_factories[entrypoint_key]
+            factory = load_module(entrypoint_key)
+            if not factory:
+                logger.warning(
+                    f"Could not find `{entrypoint_key}` tracker entrypoint. Skipping..."
+                )
+                continue
         if config:
             logger.info(f"Tracker config found for `{entrypoint_key}` as `{config}`")
-            tracker = factory(config)
         else:
             logger.info(f"No tracker config specified for `{entrypoint_key}`")
-            tracker = factory(None)
```

Comment:
was this intentional? this is technically BC breaking since `config = {}` (empty map) also evaluates to `False`. In the previous logic, this would end up calling the factory with `None` and in the new logic it would call it with `{}`.

Not a big deal since an empty-map won't cause null check errors, but FYI. Would've been a different story if `factory({})` was changed to `factory(None)` since that would've caused factories without a null-check (e.g. `if config:` or `if config is None:`) to throw.

Reply:
Not intentional, I don't even have this locally, weird. I'll make sure it's not here and no BC is introduced.